src/components/datagrid/DataGrid.js

```diff
@@ -236,7 +236,7 @@ export default class DataGridComponent extends NestedArrayComponent {
 
   render() {
     const columns = this.getColumns();
-    const colWidth = Math.floor(12 / (columns.length + 1)).toString();
+    const layoutFixed = columns.some(col => col.type === 'select') || this.component.layoutFixed;
```

Comment:
So `select`s certainly _can_ be added to a table without fixed layout without being broken.  This seems like either a fix to a particular use case or further breaking which started in https://github.com/formio/formio.js/pull/3544 but this breaks perfectly working forms from 4.12

Reply:
I agree that we should look into this. We will investigate.

Reply:
@mikekotikov @travist 
This resulted in a prod issue for us :(
If someone has a select component inside a datagrid, the fixed layout is making the datagrid look weird. 
![image](https://user-images.githubusercontent.com/4932306/121134527-1f6bd980-c851-11eb-8df5-297cdd4bc3ab.png)